src/libs/elektra/kdb.c

```diff
@@ -505,24 +506,26 @@ static int elektraGetCheckUpdateNeeded (Split * split, Key * parentKey)
 	for (size_t i = 0; i < split->size; i++)
 	{
 		int ret = -1;
-		Backend * backend = split->handles[i];
+		Plugin * backend = split->handles[i];
 		clear_bit (split->syncbits[i], (splitflag_t) SPLIT_FLAG_SYNC);
 
-		Plugin * resolver = backend->getplugins[RESOLVER_PLUGIN];
-		if (resolver && resolver->kdbGet)
+		if (backend && backend->kdbGet)
 		{
 			ksRewind (split->keysets[i]);
 			keySetName (parentKey, keyName (split->parents[i]));
 			keySetString (parentKey, "");
-			ret = resolver->kdbGet (resolver, split->keysets[i], parentKey);
+			ret = backend->kdbGet (backend, split->keysets[i], parentKey);
```

Comment:
This should not do a full `kdbGet`. You can either let this code access the `resolver` directly (like it was before) or you (probably better) set a special metakey on `parentKey` (e.g. `backend/resolveronly`) to tell the `backend` plugin that is should only call the `resolver` and nothing else.

AFAIK the purpose of this call here is to:
1. let the resolver find the filename for a mountpoint and save it in the split
2. Detect if the file exists at all. If the file doesn't exist `resolver` returns `ELEKTRA_PLUGIN_STATUS_NO_UPDATE` and we can ignore this mountpoint in `kdbGet`.
3. Detect if the cache can be used (not 100% sure how this works)

@mpranj probably knows more, since he had to adapt this to for the cache

Reply:
Good catch, @kodebach already said what is important. This should not be a full kdbGet.

The resolver is called first to determine whether we need to call the storage plugin at all. (obviously it also resolves the filenames etc.) The resolver already has similar optimizations without the cache, it's just that the cache is able to persist this information. The default resolver simply remembers the modification timestamp of the file, but this could be different for other resolvers.